types/akamai-edgeworkers/index.d.ts

```diff
@@ -0,0 +1,296 @@
+
+        /**
+         * Object containing properties specifying the device characteristics. This
+         * value of this property will be null if the contract associated with the
+         * request does not have entitlements for EDC.
+         */
+        readonly device: Device | undefined;
+
+        /**
+         * The cpcode used for reporting.
+         */
+        readonly cpCode: number;
+    }
+
+    interface MutableRequest extends MutatesHeaders, ReadsHeaders, ReadsVariables, Request {
+    }
+
+    interface ImmutableRequest extends ReadsHeaders, ReadsVariables, Request {
+    }
+
+    interface Response extends HasStatus, MutatesHeaders, ReadsHeaders {
+    }
+
+    /**
+     * Notes:
+     * * If the IP address is in the reserved IP space (as designated by the
+     *   Internet Assigned Numbers Authority), every property will have the
+     *   value of ‘reserved’.
+     * * If user location properties can not be supplied for any reason,
+     *   undefined is returned for that property
+     */
+    interface UserLocation {
+        /**
+         * The continent value is a two-letter code for the continent that
+         * the IP address maps to.
+         */
+        continent: string | undefined;
+
+        /**
+         * The country value is an ISO-3166, two-letter code for the country
+         * where the IP address maps to.
+         */
+        country: string | undefined;
+
+        /**
+         * The region value is an ISO-3166, two-letter code for the state,
+         * province, or region where the IP address maps to.
+         */
+        region: string | undefined;
+
+        /**
+         * The city value is the city (within a 50-mile radius) that the IP
+         * address maps to.
+         */
+        city: string | undefined;
+
+        /**
+         * The zipCode value is the zipcode that the IP address maps to
+         * (multiple values possible).
+         *
+         * Contiguous zip codes will be represented as a range of the form
+         * "FirstZipInRange LastZipInRange", and multiple ranges may be
+         * present (each range separated by the plus (+) character).
+         *
+         * For example, the following strings are all valid zipCode values:
+         *
+         * * 10001
+         * * 10001+10003
+         * * 10001-10003+10005
+         * * 10001-10003+10005-10008
+         *
+         * For country = US and country = PR, zip refers to the 5 digit
+         * zipcode.
+         *
+         * For country = CA, zip refers to the forward sortation area (FSA).
+         * For more information on FSA, go to http://www.canadapost.ca and
+         * search for FSA.
+         *
+         * See the EdgeScape Users Guide for more details.
+         */
+        zipCode: string | undefined;
+    }
+
+    /**
+     * Notes:
+     * * If device properties can not be supplied for any reason,
+     *   undefined is returned for each property
+     */
+    class Device {
```

Comment:
From what I can gather, the `EW` namespace in this file doesn't actually exist at runtime, and is provided solely as a place to define the types needed for the four methods mentioned in the README (and at https://developer.akamai.com/api/web_performance/edgeworkers/v1.html#edgeworkersbundleformat). By defining `Device` as a `class`, that changes the `EW` namespace from a "type-only" namespace to a "value" namespace (since classes are both types *and* values), meaning that `EW` will appear to exist as a valid expression. I would recommend changing `Device` to an interface to avoid this:

```suggestion
    interface Device {
```

Reply:
You're correct - I've made the change. Thanks for the background.